### ISSUE TITLE
Does not work with components that use angular2-html-loader syntax

### ISSUE THREAD

**Issue by jusefb** [thymikee/jest-preset-angular#28]
I can't get this to work with components that use angular2-html-loader. The issue is with the templateUrl: property syntax. angular2-html-loader allows to specify a template using the following syntax without the "./" which would still resolve to the same location as the component:

```javascript
templateUrl: 'template.html'
```

however, jest-prest-angular only works if the template has a ./ infront of it. I have tried to change the preprocessor login to this:

```javascript
const {process} = require('ts-jest/preprocessor.js');
const TEMPLATE_URL_REGEX = /templateUrl:\s*(?:")(.*)(?:")/g;
const STYLE_URLS_REGEX = /styleUrls:\s*\[\s*((?:'|").*\s*(?:'|")).*\s*.*\]/g;

module.exports.process = (src, path, config) => {
  // Replace `templateUrl: ''` calls with `template: require('')`
  // and `styleUrls: ['']` with `styles: []`

  return process(
    src
      .replace(TEMPLATE_URL_REGEX, 'template: require("./$1")')
      .replace(STYLE_URLS_REGEX, 'styles: []'),
    path,
    config
  );
};
```
effectively truing to add the missing ./ to the templateUrl but it does not seem to work. 

Any ideas?

**Comment by thymikee** [comment]
I'm working on AST transform instead of regex, so I can address issues like this. But technically, if `template.html` is adjacent to your test file, it should work with adding `./`, even for relative paths.

**Comment by jusefb** [comment]
Thank you for a quick response. I get the following error in my component when using the code above:

```
    SyntaxError

      at XMLHttpRequest.open (..\..\..\..\..\..\..\AppData\Roaming\npm\node_modules\jest\node_modules\jsdom\lib\jsdom\li
ving\xmlhttprequest.js:486:15)
      at ResourceLoaderImpl.Object.<anonymous>.ResourceLoaderImpl.get (node_modules\@angular\platform-browser-dynamic\bu
ndles\platform-browser-dynamic.umd.js:42:13)
      at DirectiveNormalizer.Object.<anonymous>.DirectiveNormalizer._fetch (node_modules\@angular\compiler\bundles\compi
ler.umd.js:12779:43)
      at DirectiveNormalizer.Object.<anonymous>.DirectiveNormalizer.normalizeTemplateAsync (node_modules\@angular\compil
er\bundles\compiler.umd.js:12831:21)
      at DirectiveNormalizer.Object.<anonymous>.DirectiveNormalizer.normalizeTemplate (node_modules\@angular\compiler\bu
ndles\compiler.umd.js:12803:44)
      at CompileMetadataResolver.Object.<anonymous>.CompileMetadataResolver._loadDirectiveMetadata (node_modules\@angula
r\compiler\bundles\compiler.umd.js:13722:75)
      at node_modules\@angular\compiler\bundles\compiler.umd.js:13915:54
      at Array.forEach (native)
      at CompileMetadataResolver.Object.<anonymous>.CompileMetadataResolver.loadNgModuleDirectiveAndPipeMetadata (node_m
odules\@angular\compiler\bundles\compiler.umd.js:13914:41)
      at node_modules\@angular\compiler\bundles\compiler.umd.js:25059:58
```

Can you see what I am doing wrong?

**Comment by thymikee** [comment]
Change `(?:")` to `(?:"|')`

**Comment by jusefb** [comment]
Thanks, I have maanged to resolve the error with the template but I am still not able to run tests that have dependencies. Just keep getting an obscure syntax error whenever I try to inject dependencies into the test:

```javascript
    SyntaxError

      at XMLHttpRequest.open (..\..\..\..\..\..\..\AppData\Roaming\npm\node_modules\jest\node_modules\jsdom\lib\jsdom\li
ving\xmlhttprequest.js:486:15)
      at ResourceLoaderImpl.Object.<anonymous>.ResourceLoaderImpl.get (node_modules\@angular\platform-browser-dynamic\bu
ndles\platform-browser-dynamic.umd.js:42:13)
      at DirectiveNormalizer.Object.<anonymous>.DirectiveNormalizer._fetch (node_modules\@angular\compiler\bundles\compi
ler.umd.js:12779:43)
      at DirectiveNormalizer.Object.<anonymous>.DirectiveNormalizer.normalizeTemplateAsync (node_modules\@angular\compil
er\bundles\compiler.umd.js:12831:21)
      at DirectiveNormalizer.Object.<anonymous>.DirectiveNormalizer.normalizeTemplate (node_modules\@angular\compiler\bu
ndles\compiler.umd.js:12803:44)
      at CompileMetadataResolver.Object.<anonymous>.CompileMetadataResolver._loadDirectiveMetadata (node_modules\@angula
r\compiler\bundles\compiler.umd.js:13722:75)
      at node_modules\@angular\compiler\bundles\compiler.umd.js:13915:54
      at Array.forEach (native)
      at CompileMetadataResolver.Object.<anonymous>.CompileMetadataResolver.loadNgModuleDirectiveAndPipeMetadata (node_m
odules\@angular\compiler\bundles\compiler.umd.js:13914:41)
      at node_modules\@angular\compiler\bundles\compiler.umd.js:25059:58
```

**Comment by thymikee** [comment]
Just released v1.2 with the fix.
See working example: https://github.com/thymikee/jest-preset-angular/blob/master/example/src/app/simple/simple.component.ts

**Comment by jusefb** [comment]
Yup, that works. Thanks man keep up the awesome work you are doing.
